package FluidNexus;

option java_package = "net.fluidnexus.FluidNexus";
option java_outer_classname = "FluidNexusProtos";

//"create table FluidNexusData (_id integer primary key autoincrement,    source varchar(32), time bigint, type integer, title varchar(40), data long     varchar, hash varchar(32), cellID varchar(20), mine bit);";

message FluidNexusHashes {
    repeated string hash = 1;
}

message FluidNexusMessage {
    optional string title = 1;
    optional string message = 2;
    optional float timestamp = 3;
    optional string hash = 4;
    optional string source = 5;
    optional bool mine = 6;

    // Do we want to do it this way, or use mime-types?
    enum MessageType {
        TEXT = 0;
        IMAGE = 1;
        AUDIO = 2;
        VIDEO = 3;
    }

    optional bytes attachment = 7;
}

message FluidNexusMessages {
    repeated FluidNexusMessage message = 1;
}
